clang/runtime/dpct-rt/include/dpct/graph.hpp

```diff
@@ -89,5 +91,47 @@ static inline bool end_recording(sycl::queue *queue_ptr,
   return detail::graph_mgr::instance().end_recording(queue_ptr, graph);
 }
 
+/// Adds an empty node to the command graph with optional
+/// dependencies.
+/// \param [out] newNode A pointer to the node_ptr that will be
+/// added to the graph.
+/// \param [in] graph A pointer to the command graph.
+/// \param [in] dependenciesArray An array of node pointers
+/// representing the dependencies of the new node.
+/// \param [in] numberOfDependencies The number of dependencies in
+/// the dependenciesArray.
+static void add_empty_node(dpct::experimental::node_ptr *newNode,
+                           dpct::experimental::command_graph_ptr graph,
+                           dpct::experimental::node_ptr *dependenciesArray,
+                           std::size_t numberOfDependencies) {
+  *newNode =
+      std::make_shared<sycl::ext::oneapi::experimental::node>(graph->add());
+  if (numberOfDependencies) {
```

Comment:
Can we use one property `depends_on` which contains all the dependencies, or add multiple properties in one call?

Reply:
Added logic to add dependencies at once.